cpp/src/Ice/ThreadPool.cpp

```diff
@@ -1233,6 +1218,13 @@ IceInternal::ThreadPool::EventHandlerThread::setState(Ice::Instrumentation::Thre
     _state = s;
 }
 
+void
+IceInternal::ThreadPool::EventHandlerThread::start()
+{
+    assert(_thread.get_id() == thread::id()); // Not started yet
+    _thread = thread(&EventHandlerThread::run, this);
```

Comment:
It would be nice to add a comment about the shared ownnership of "this". How do we ensure this is not deleted while the thread is running?

Reply:
I will add a comment, the thread pool holds a shared ptr to the EventHandlerThread object  and join the threads before exiting.